### PR TITLE
Generate user queries in GraphQL tests

### PR DESCRIPTION
**Pull request by chrzaszcz**
Before the change, admin operations used auto-generated queries, while user operations used manually coded GraphQL documents.

After the change, the `build_specs` function is called with RPC per test group. It auto-generates all queries for the given endpoint. As a byproduct, the query generation code is better tested.

The specs are stored in a persistent term. There was an attempt to store them in `Config`, but the specs are huge, and test reports would grow in size and become less readable.

The scope of this PR includes the `account`, `metrics` and `vcard` categories.

Also:
- Support empty argument lists in query generation. This corner case was revealed by the `unregister` operation.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/129820) / small_tests / 803f170dfa0d86414f5f547e2388fc91d904e135
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3709/129820/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3709/129820/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-07-14_14.16.48/lib.mongooseim.logs/run.2022-07-14_14.16.48/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/129850) / small_tests / a16920418ece0a72256140a62843239d35d95655
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3709/129850/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3709/129850/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-07-15_06.58.24/lib.mongooseim.logs/run.2022-07-15_06.58.25/suite.log.html)

**Review by JanuszJakubiec — APPROVED** [review]
looks good

File: big_tests/tests/graphql_vcard_SUITE.erl [hunk 1]
```diff
@@ -362,114 +350,3 @@ agent_vcard_input() ->
...
-    ResultFormat = get_full_vcard_as_result(),
-    <<"mutation M1($vcard: VcardInput!){vcard{setVcard(vcard: $vcard)",
-      ResultFormat/binary, "}}">>.
-
-user_get_full_vcard_as_result_query() ->
-    ResultFormat = get_full_vcard_as_result(),
-    <<"query Q1($user: JID){vcard{getVcard(user: $user)",
-      ResultFormat/binary, "}}">>.
-
-get_full_vcard_as_result() ->
```
Comment (Premwoik) [inline big_tests/tests/graphql_vcard_SUITE.erl:393]: Very nice that we manage to remove this big query :D

**Review by Premwoik — APPROVED** [review]
All good!
